### DISCUSSION TITLE
How GNPy Implements Flex Grid Transmission Simulation？

### DISCUSSION THREAD

**Discussion by ZYIAO** [Telecominfraproject/oopt-gnpy#465] [Q&A] [answered]
Thank you for providing the useful tool. As mentioned in the title, I would like to ask if GNPy can simulate flex-grid transmission in end-to-end links, and even in large-scale networks. Also, if possible, please provide some usage assistance. Thank you again!

**Comment by jktjkt** [comment, marked as answer]
Hi @ZYIAO , yes. All the information that are needed should be available from [our documentation](https://gnpy.readthedocs.io/en/master/). You'll need the latest release in which we [enabled defining the custom spectrum](https://gnpy.readthedocs.io/en/master/json.html#arbitrary-channel-definition) by the user without coding in Python.

**Comment by ZYIAO** [comment]
Thank you very much for reminding me that the latest version is ready to do this!
